### PR TITLE
bug fix: generate directory if path not exist in compile_mmap_model

### PR DESCRIPTION
**Pull request by jianhao2016**
*Issue #, if available:* N/A

*Description of changes:*
Generate directory if path not exist when saving a XLinearModel. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pecos/xmc/xlinear/model.py [hunk 1]
```diff
@@ -144,6 +144,8 @@ def compile_mmap_model(cls, npz_folder, mmap_folder):
         """
         import shutil
 
+        if not path.exists(mmap_folder):
```
Comment (weiliw-amz) [inline pecos/xmc/xlinear/model.py]: How about using `os.makedirs(mmap_folder, exist_ok=True)`?
https://docs.python.org/3/library/os.html#os.makedirs

Comment (jianhao2016) [reply pecos/xmc/xlinear/model.py]: I was following the convention of previous changes, in that case should we change https://github.com/amzn/pecos/blob/mainline/pecos/xmc/xlinear/model.py#L98 as well?

**Review by OctoberChang — APPROVED** [review]
LGTM

File: pecos/xmc/xlinear/model.py (hunk 1 above)
Comment (jianhao2016) [reply pecos/xmc/xlinear/model.py]: Updated both lines. Please check revision.

**Review by weiliw-amz — APPROVED** [review]

**Pull request merged**